File: packages/java/src/main/java/io/tokid/TokidSdk.java
package io.tokid;

import java.util.List;

public final class TokidSdk {
  private TokidSdk() {}

  public static String defaultProfileId() {
    return TokidKernel.defaultKernel().defaultProfileId();
  }

  public static int defaultTokidLength() {
    return TokidKernel.defaultKernel().profile(defaultProfileId()).entropy().recommendedLength();
  }

  public static TokidFactory createFactory(String profileId, Integer length) {
    return new TokidFactory(profileId, length, null);
  }

  public static String generate() {
    return generate(null, null, null);
  }

  public static String generate(String profileId, Integer length, String format) {
    return createFactory(profileId, length).generate(length, format);
  }

  public static Tokid parse(String value, String profileId) {
    return TokidKernel.defaultKernel().parseWithCodec(value, profileId, null);
  }

  public static boolean isTokid(String value, String profileId) {
    return TokidKernel.defaultKernel().validateTokid(value, profileId);
  }

  public static String toPrompt(Object value, String profileId) {
    return TokidKernel.defaultKernel().renderTokid(resolveTokid(value, profileId), "prompt");
  }

  public static String toTransport(Object value, String profileId) {
    return TokidKernel.defaultKernel().renderTokid(resolveTokid(value, profileId), "transport");
  }

  public static String toEnvelope(Object value, String profileId) {
    return TokidKernel.defaultKernel().renderTokid(resolveTokid(value, profileId), "envelope");
  }

  public static List<String> listProfiles() {
    return TokidKernel.defaultKernel().listProfileIds();
  }

  static Tokid resolveTokid(Object value, String profileId) {
    if (value instanceof Tokid tokid) {
      return tokid;
    }
    if (value instanceof String rendered) {
      Tokid parsed = parse(rendered, profileId);
      if (parsed == null) {
        throw new IllegalArgumentException("Invalid tokid value");
      }
      return parsed;
    }
    throw new IllegalArgumentException("Unsupported tokid input");
  }
}

File: packages/dotnet/Tokid/TokidSdk.cs
using System.Collections.Generic;

namespace Tokid
{
    public static class TokidSdk
    {
        public static string DefaultProfileId => TokidKernel.Default.DefaultProfileId;
        public static int DefaultTokidLength => TokidKernel.Default.Profile(DefaultProfileId).Entropy.RecommendedLength;

        public static TokidFactory CreateFactory(string profileId = null, int? length = null)
        {
            return new TokidFactory(profileId, length);
        }

        public static string Generate(string profileId = null, int? length = null, string format = "envelope")
        {
            return CreateFactory(profileId, length).Generate(length, format);
        }

        public static Tokid Parse(string value, string profileId = null)
        {
            return TokidKernel.Default.ParseWithCodec(value, profileId, null);
        }

        public static bool IsTokid(string value, string profileId = null)
        {
            return TokidKernel.Default.ValidateTokid(value, profileId);
        }

        public static string ToPrompt(object value, string profileId = null)
        {
            return TokidKernel.Default.RenderTokid(ResolveTokid(value, profileId), "prompt");
        }

        public static string ToTransport(object value, string profileId = null)
        {
            return TokidKernel.Default.RenderTokid(ResolveTokid(value, profileId), "transport");
        }

        public static string ToEnvelope(object value, string profileId = null)
        {
            return TokidKernel.Default.RenderTokid(ResolveTokid(value, profileId), "envelope");
        }

        public static IReadOnlyList<string> ListProfiles()
        {
            return TokidKernel.Default.ListProfileIds();
        }

        private static Tokid ResolveTokid(object value, string profileId)
        {
            var tokid = value as Tokid;
            if (tokid != null)
            {
                return tokid;
            }

            var rendered = value as string;
            if (rendered != null)
            {
                var parsed = Parse(rendered, profileId);
                if (parsed == null)
                {
                    throw new System.ArgumentException("Invalid tokid value");
                }
                return parsed;
            }

            throw new System.ArgumentException("Unsupported tokid input");
        }
    }
}
